### PR TITLE
Fix according to last comments and add table-driven tests

### PR DESCRIPTION
**Pull request by CapUlDis**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
-def hello():
-    return "Hello World!"
-
-@app.route('/login', methods=['POST'])
-def login():
+def create_app():
+    app = Flask(__name__)
     
-    data = request.get_json(force=True)
+    @app.route('/login', methods=['POST'])
```
Comment (nkartashov) [inline main.py]: You probably don't want to define the method inside `create_app`, do it like you did it before.

File: main.py [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+    def login():
+        
+        if 'Content-Type' not in request.headers.keys() or request.headers['Content-Type'] != 'application/json':
+            http_code = 400
```
Comment (nkartashov) [inline main.py]: There is no need for a separate variable for http code, inline it, please. Applies throughout.

File: main.py [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+            return 'Data is not in json format.', http_code
+                
+        if 'name' not in data.keys() or 'password' not in data.keys():
+            http_code = 400
+            return 'Name or/and password are missing.', http_code
+        
+        if data['name'] == 'denchik' and data['password'] == 'foobar':
+            http_code = 200
+            return 'Correct name and password.', http_code
+        else:
```
Comment (nkartashov) [inline main.py]: You don't need an `else` if the previous block ends with `return`.

File: main.py [hunk 4]
```diff
@@ -1,22 +1,37 @@
...
-@app.route('/login', methods=['POST'])
-def login():
+def create_app():
+    app = Flask(__name__)
     
-    data = request.get_json(force=True)
+    @app.route('/login', methods=['POST'])
+    def login():
+        
+        if 'Content-Type' not in request.headers.keys() or request.headers['Content-Type'] != 'application/json':
```
Comment (nkartashov) [inline main.py]: Please use `request.get_json()`
https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.get_json

File: tests/test_post_login.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import pytest
+import json
+
+@pytest.mark.parametrize(
+        ('datam', 'header', 'message'),
```
Comment (nkartashov) [inline tests/test_post_login.py]: What does `datam` mean?

File: tests/test_post_login.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+                (json.dumps({"name": "denchik", "password": "foo"}), {'content-type': 'application/json'}, b'Wrong name or password.'),
+                
+        ),
+)
+def test_login_with_different_data_headers(client, app, datam, header, message):
+    response = client.post(
+            '/login', data = datam, headers = header
+            )
+    
+    assert message in response.data
```
Comment (nkartashov) [inline tests/test_post_login.py:23]: Please add an http code assertion.

File: tests/test_post_login.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        (
+                ('', '', b'Content-Type is not json.'),
+                ('', {'content-type': 'xml'}, b'Content-Type is not json.'),
+                ('smthg', {'content-type': 'application/json'}, b'Data is not in json format.'),
+                (json.dumps({"exp1": "exp2", "exp3": "exp4"}), {'content-type': 'application/json'}, b'Name or/and password are missing.'),
+                (json.dumps({"name": "denchik", "password": "foobar"}), {'content-type': 'application/json'}, b'Correct name and password.'),
+                (json.dumps({"name": "den", "password": "foobar"}), {'content-type': 'application/json'}, b'Wrong name or password.'),
+                (json.dumps({"name": "denchik", "password": "foo"}), {'content-type': 'application/json'}, b'Wrong name or password.'),
+                
+        ),
```
Comment (nkartashov) [inline tests/test_post_login.py:15]: Great work!

File: tests/test_post_login.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+)
```
Comment (nkartashov) [inline tests/test_post_login.py:17]: Please add ids to the tests (`ids` argument) for better readability.
https://docs.pytest.org/en/latest/reference.html#pytest-mark-parametrize-ref

**Review by nkartashov — CHANGES_REQUESTED** [review]
Great job!

File: main.py (hunk 2 above)
Comment (CapUlDis) [reply main.py]: Done!

File: tests/test_post_login.py (hunk 6 above)
Comment (CapUlDis) [reply tests/test_post_login.py:23]: Done!

Comment (CapUlDis) [reply tests/test_post_login.py:23]: Done!

File: tests/test_post_login.py (hunk 5 above)
Comment (CapUlDis) [reply tests/test_post_login.py]: Done!

File: main.py (hunk 2 above)
Comment (CapUlDis) [reply main.py]: Done!

File: tests/test_post_login.py (hunk 8 above)
Comment (CapUlDis) [reply tests/test_post_login.py:17]: Done!

File: main.py (hunk 3 above)
Comment (CapUlDis) [reply main.py]: Done!

File: main.py (hunk 4 above)
Comment (CapUlDis) [reply main.py]: Done!

File: tests/conftest.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import pytest
+
+from main import create_app, login
+
+
+@pytest.fixture
+def app():
+    app = create_app()
+    app.add_url_rule('/login', view_func=login, methods=['POST'])
```
Comment (nkartashov) [inline tests/conftest.py]: This belongs to main.py, not in tests.

**Review by nkartashov — CHANGES_REQUESTED** [review]

**Pull request merged**